scipy/linalg/_matfuncs_expm.pyx.in

```diff
@@ -339,6 +358,11 @@ cdef void pade_9_UV_calc_{{ pfx }}({{ tname }}[:, :, ::]Am, int n) noexcept nogi
         b[7] = 3960.
         b[8] = 90.
 
+        # Compute this in case we skipped it before
+        if n >= 400:
+            {{ pfx }}gemm(cN, cN, &n, &n, &n, &one, &Am[2, 0, 0], &n,
```

Comment:
You can just use 

```python
            np.matmul(Am[2, :, :], Am[2, :, :], out=Am[4, :, :])
```

Reply:
This would be so much cleaner. I tried it but got an error related to the `nogil` modifier. I'm not super familiar with Cython and don't know how to work around this.

Reply:
Ah yes I forgot the nogil part then nevermind it is good as is